dask/imperative.py

```diff
@@ -63,6 +63,12 @@ def to_task_dasks(expr):
     """
     if isinstance(expr, Value):
         return expr.key, expr._dasks
+    elif isinstance(expr, base.Base):
+        name = tokenize(str(expr), True)
+        keys = expr._keys()
+        dsk = expr._optimize(expr.dask, keys)
+        dsk[name] = (expr._finalize, expr, (list, keys))
```

Comment:
Beware that `keys` may be a list of lists.  There is probably a `concrete` function lying around that might replace `list` here.